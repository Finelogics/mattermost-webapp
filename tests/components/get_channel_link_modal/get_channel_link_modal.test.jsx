// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import GetChannelLinkModal from 'components/get_channel_link_modal/get_channel_link_modal.jsx';
import GetLinkModal from 'components/get_link_modal.jsx';

// mock the site URL, as it will be prefixed onto the link
jest.mock('utils/url', () => {
    return {
        getSiteURL: jest.fn(() => 'http://localhost:8065')
    };
});

describe('components/GetChannelLinkModal', () => {
    test('should match snapshot with currentTeamUrl passed in', () => {
        const wrapper = shallow(
            <GetChannelLinkModal
                link='/foo/bar'
            />
        );

        expect(wrapper).toMatchSnapshot();
        const helpText = 'Paste the link below into a web browser to view the channel.';
        expect(wrapper.find(GetLinkModal).prop('helpText')).toEqual(helpText);
        expect(wrapper.find(GetLinkModal).prop('show')).toEqual(false);
        expect(wrapper.find(GetLinkModal).prop('title')).toEqual('Channel Link');
        expect(wrapper.find(GetLinkModal).prop('link')).toEqual('http://localhost:8065/foo/bar');

        wrapper.setState({post: {id: 'post_id'}});
        expect(wrapper.find(GetLinkModal).prop('link')).toEqual('http://localhost:8065/foo/bar');
    });

    test('should call hide on GetLinkModal\'s onHide', () => {
        const wrapper = shallow(
            <GetChannelLinkModal/>
        );

        wrapper.find(GetLinkModal).first().props().onHide();
        expect(wrapper.state('show')).toBe(false);
    });
});
