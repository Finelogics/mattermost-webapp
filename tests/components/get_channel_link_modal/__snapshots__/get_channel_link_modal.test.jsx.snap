// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GetChannelLinkModal should match snapshot with currentTeamUrl passed in 1`] = `
<GetLinkModal
  helpText="Paste the link below into a web browser to view the channel."
  link="http://localhost:8065/foo/bar"
  onHide={[Function]}
  show={false}
  title="Channel Link"
/>
`;
